templates/ingress.yaml

```diff
@@ -409,3 +409,38 @@ spec:
   {{- end }}
 {{- end }}
 ---
+{{- if and (merge .Values dict | dig "coderd" "serviceNext" true) (merge .Values dict | dig "ingress" "enable" true) }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: coderd-ingress
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    nginx.ingress.kubernetes.io/proxy-body-size: "0"
+  {{- range $key, $value := merge .Values dict | dig "ingress" "annotations" dict }}
+    {{ $key }}: {{ $value | quote }}
+  {{- end }}
+spec:
+{{- include "coder.ingress.tls" . }}
+  rules:
```

Comment:
suggest we add a defaultBackend too: https://kubernetes.io/docs/concepts/services-networking/ingress/#default-backend

Reply:
Default backend is supposed to be on the controller rather than the ingress resources (by convention)... this choice was deliberate.